### ISSUE TITLE
file size and position should be u64 (not int) on 64 bit OSes

### ISSUE THREAD

**Issue by saasira** [vlang/v#7511]
**V version:** 
**OS:** 

**What did you do?**
os.file_size(filepath)

**What did you expect to see?**
8 GB for an 8 GB file
os.file_size
os.read_at
os.read_bytes_at
os.read_bytes_into
os.write_bytes_at
os.write_to
 
 and other such methods should use u64 and not int on 64 bit OSes
 
**What did you see instead?**
some garbage integer; perhaps u64 value rounded off to int after several rounds

int for file_size is severe limitation on 64 bit OSes; it can only represent 2 GB file. file size & position must be referred using unsigned 64 bit number (u64) on 64 bit OSes.

perhaps, it should use the usize that was discussed in the ticket : 
https://github.com/vlang/v/pull/6213

this discussion on Discord can provide the context
https://discord.com/channels/592103645835821068/592294828432424960/791343348728332298

**Comment by bstnbuck** [comment]
After a long search and a lot of testing, I got it working. Since several files need to be changed to u64, I have only created println here. However, these should work. Attention! Since TCC is a 32 bit compiler, it will output wrong information after 2GB. Windows 10 64 bit, Ubuntu 20.04 64 bit was tested. Below is a temporary workaround. 
1. Define following C functions for Linux and Windows
```v
fn C.lstat64(charptr, voidptr) u64 // linux 64 support
fn C._wstat64() //windows 64 bit support
```
2. create the following struct for "_wstat64()"
```v
struct C.__stat64{
	st_size  u64
	st_mode  u32
	st_mtime int
}
```
3. edit the function
```v
pub fn file_size(path string) int {
	mut s := C.stat{}

	unsafe{
		$if windows{
			$if tinyc{
				C.stat(charptr(path.str), &s)
				println("tcc win")
				println(s.st_size)
			}$else{
				$if x64{
					mut swin := C.__stat64{}
					C._wstat64(path.to_wide(), voidptr(&swin))
					println("win64")
					println(swin.st_size)
				}$else{
					C._wstat(path.to_wide(), voidptr(&s))
					println("win32")
					println(s.st_size)
				}
			}
		}$else{
			$if tinyc{
				C.stat(charptr(path.str), &s)
			}$else{
				C.lstat64(charptr(path.str), &s)
			}
			println(s.st_size)
		}
	}
	return 0
}
```
Please do not close the bug report, because it is only a workaround!

**Comment by saasira** [comment]
thanks for your effort in providing this workaround; i too attempted to get this working but then realized that this is not a one off usecase - entire os module [functions dealing with file system] must be refactored to work with 32 bit file size/position on 32 bit OS and 64 bit file size/position on 64 bit OS. this should be a language level change that the V core team should focus on rather than we the end users sharing workarounds or patches.

for now, i decided to better wait until this particular issue is fixed, since i started using V mainly for writing a key-value store [in V]; which is not possible at this moment until V handles large files without issues

**Comment by bstnbuck** [comment]
Yes! You are absolutely right. When I created the workaround, I noticed the many dependencies. I wrote a shredding program (program which deletes files safely). There it can come likewise to larger files. I have a module built in, which includes the above workaround and then imported this separately. I have adapted the workaround again. As it seems, there are only problems with the Windows TCC, because it only understands 32 bit. If you want to implement it yourself, here is the link to my workaround.
https://github.com/bstnbuck/VShred/blob/main/os_stat/f_size.v
